File: client/src/interfaces/interfaces.ts
export interface Movie {
    id: number;
    title: string;
    year: number;
    rating: number;
    description: string;
    uri: string;
}

export interface Serie {
    id: number
    title: string
    year: string
    rating: number
    description: string
    episodes: [Movie]
}

export interface Collection {
    movies: Movie[]
    series: Serie[]
}

File: server/interfaces.py
from dataclasses import dataclass
from typing import List

@dataclass
class Movie:
    id: int
    title: str
    year: str
    rating: float
    description: str
    uri: str

@dataclass
class Serie:
    id: int
    title: str
    year: str
    rating: float
    description: str
    episodes: List[Movie]

@dataclass
class Collection:
    movies: List[Movie]
    series: List[Serie]
